app/models/download.rb

```diff
@@ -0,0 +1,30 @@
+class Download < ApplicationRecord
+  belongs_to :teacher
+  belongs_to :lesson
+  has_one_attached :lesson_bundle
+
+  validates :teacher, presence: true
+  validates :lesson, presence: true
+  validates :lesson_bundle, content_type: 'application/zip'
+
+  has_many :download_transitions, autosave: false, dependent: :destroy
+
+  include Statesman::Adapters::ActiveRecordQueries[
+    transition_class: DownloadTransition,
+    initial_state: :pending
+  ]
+
+  def state_machine
+    @state_machine ||= DownloadStateMachine.new(self, transition_class: DownloadTransition)
+  end
+
+  delegate :transition_to!, :current_state, to: :state_machine
+
+  def completed?
+    current_state == 'completed'
+  end
```

Comment:
It might be worth delegating `#in_state?` to `state_machine`. Then instead of comparing with `==` you can say
```ruby
def completed?
  in_state?(:completed)
end
```